new/mode-agnostic-data-collection.md

```diff
@@ -0,0 +1,560 @@
+---
+RFC: unassigned
+Author: Adam Leff <adamleff@chef.io>
+Status: Draft
+Type: Standards Track
+---
+
+# Mode-Agnostic Run Data Collection
+
+In order to provide increased clarity into a user's fleet operations, and to support Chef deployments that do not necessarily include the use of a Chef Server, this change aims to provide an easy and consistent way for the Chef Client to report run statistics to a data collection system. The initial statistics we look to collect include node state, converge start/end times, and converge details.
```

Comment:
We should better differentiate between the tool, **Chef** and the modes, e.g. **Chef Client** and **Chef Solo**. There's a clear bias in this document for the world in which Chef is a client, which reads strange in some examples here.

Can we agree on some definitions, maybe put them somewhere?

Chef - The tool you use to automate your system.
Chef Client [Mode] - when using Chef in client mode
Chef Solo [Mode] - when using Chef in solo Mode in >= 12.10.54 / post RFC 031
Chef Solo Legacy Mode - when using Chef in solo mode in < 12.10.54 / pre RFC 031, or using `--legacy-mode`.

Chef Local Mode - Mostly the same as Chef Solo Mode in 12.10.54 / post RFC 031?